src/modules/patients/patients.service.ts

```diff
@@ -31,17 +32,35 @@ export class PatientsService {
         HttpStatus.BAD_REQUEST,
       );
     }
-    const patien = this.patientsRepository.create(createPatientInput);
-    if (userId && !createUserInput) {
-      patien.userId = userId;
-    }
-    if (createUserInput && !userId) {
-      const user = await this.usersService.create(createUserInput);
-      patien.userId = user.id;
+    try {
+      const patien = this.patientsRepository.create(createPatientInput);
+      if (userId && !createUserInput) {
+        patien.userId = userId;
+      }
+      if (createUserInput && !userId) {
+        const user = await this.usersService.create(createUserInput);
+        patien.userId = user.id;
+      }
+
+      return await this.patientsRepository.save(patien);
+    } catch (err) {
+      throw new HttpException(err, HttpStatus.BAD_REQUEST);
     }
+  }
 
+  edit(
+    patientId: string,
+    editPatientInput: EditPatientInput,
+  ): Promise<Patient> {
     try {
-      return await this.patientsRepository.save(patien);
+      if (Object.keys(editPatientInput).length === 0) {
+        throw new HttpException(
+          'You must pass at least one field to update',
```

Comment:
makes no sense i think, nothing will happen if not fields provided, but its ok

Reply:
it crashing the server when editPatientInput come as an empty object